@nx/remix - vitest keywords not found, pnpm triggers ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL for 'serve' target

### Current Behavior

in newly generated nx monorepo with a single remix app, the file at tests/routes/_index.spec.tsx triggers a typescript error saying it cannot find name 'test'. furthermore, when attempting to run the 'serve' target with pnpm, which default generates to `pnpm exec remix-serve build/index.js`, it triggeres an `ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL Command "remix-serve" not found` error.

### Expected Behavior

the types included in the tsconfig.spec.ts should be respected in the test example file and i shouldn't have to swap out pnpm for npx (which works as a fill in replacement) to get the serve target to work.

### GitHub Repo

_No response_

### Steps to Reproduce

1. pnpm dlx create-nx-workspace --packageManager="pnpm"
2. pnpm add -D @nx/remix
3. generate a remix app (id also point out it fails to perform dry runs)
4. check typescript error that generates in tests file
5. run build target, then attempt to run serve target.

### Nx Report

```shell
Node   : 20.9.0
   OS     : win32-x64
   pnpm   : 8.14.0

   nx                 : 17.2.8
   @nx/js             : 17.2.8
   @nx/linter         : 17.2.8
   @nx/eslint         : 17.2.8
   @nx/workspace      : 17.2.8
   @nx/devkit         : 17.2.8
   @nx/eslint-plugin  : 17.2.8
   @nx/react          : 17.2.8
   @nx/remix          : 17.2.8
   @nrwl/tao          : 17.2.8
   @nx/vite           : 17.2.8
   @nx/web            : 17.2.8
   typescript         : 5.3.3
```


### Failure Logs

```shell
💿  remix dev
 info  building...
 info  built (733ms)
'remix-serve' is not recognized as an internal or external command,
operable program or batch file.
undefined
 ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command "remix-serve" not found
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^
Error: connect ECONNREFUSED 127.0.0.1:4200
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1595:16) {
  errno: -4078,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 4200
}
Node.js v20.9.0
```


### Package Manager Version

pnpm - 8.14.0

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/packages/remix/src/generators/application/application.impl.spec.ts
+++ b/packages/remix/src/generators/application/application.impl.spec.ts
@@ -323,6 +323,6 @@ function expectTargetsToBeCorrect(tree: Tree, projectRoot: string) {
   expect(targets.start.command).toEqual('remix-serve build/index.js');
   expect(targets.start.options.cwd).toEqual(projectRoot);
   expect(targets.typecheck).toBeTruthy();
-  expect(targets.typecheck.command).toEqual('tsc');
+  expect(targets.typecheck.command).toEqual('tsc --project tsconfig.app.json');
   expect(targets.typecheck.options.cwd).toEqual(projectRoot);
 }

--- a/packages/remix/src/generators/application/application.impl.ts
+++ b/packages/remix/src/generators/application/application.impl.ts
@@ -78,7 +78,7 @@ export default async function (tree: Tree, _options: NxRemixGeneratorSchema) {
         },
       },
       typecheck: {
-        command: `tsc`,
+        command: `tsc --project tsconfig.app.json`,
         options: {
           cwd: options.projectRoot,
         },

--- a/packages/remix/src/generators/application/files/common/tsconfig.app.json__tmpl__
+++ b/packages/remix/src/generators/application/files/common/tsconfig.app.json__tmpl__
@@ -1,0 +1,20 @@
+{
+  "extends": "./tsconfig.json",
+  "include": [
+    "remix.env.d.ts",
+    "app/**/*.ts",
+    "app/**/*.tsx",
+    "app/**/*.js",
+    "app/**/*.jsx"
+  ],
+  "exclude": [
+    "tests/**/*.spec.ts",
+    "tests/**/*.test.ts",
+    "tests/**/*.spec.tsx",
+    "tests/**/*.test.tsx",
+    "tests/**/*.spec.js",
+    "tests/**/*.test.js",
+    "tests/**/*.spec.jsx",
+    "tests/**/*.test.jsx"
+  ]
+}

--- a/packages/remix/src/generators/application/files/common/tsconfig.json__tmpl__
+++ b/packages/remix/src/generators/application/files/common/tsconfig.json__tmpl__
@@ -1,18 +1,24 @@
 {
   "extends": "<%= offsetFromRoot %>tsconfig.base.json",
-  "include": ["remix.env.d.ts", "**/*.ts", "**/*.tsx", "**/*.js", "**/*.jsx"],
   "compilerOptions": {
-    "lib": ["DOM", "DOM.Iterable", "ES2019"],
-    "isolatedModules": true,
-    "esModuleInterop": true,
-    "jsx": "react-jsx",
-    "moduleResolution": "node",
-    "resolveJsonModule": true,
-    "target": "ES2019",
-    "strict": true,
-    "allowJs": true,
-    "forceConsistentCasingInFileNames": true,
-    // Remix takes care of building everything in `remix build`.
-    "noEmit": true
-  }
+      "lib": ["DOM", "DOM.Iterable", "ES2019"],
+      "isolatedModules": true,
+      "esModuleInterop": true,
+      "jsx": "react-jsx",
+      "moduleResolution": "node",
+      "resolveJsonModule": true,
+      "target": "ES2019",
+      "strict": true,
+      "allowJs": true,
+      "forceConsistentCasingInFileNames": true,
+      // Remix takes care of building everything in `remix build`.
+      "noEmit": true
+    },
+    "include": [],
+    "files": [],
+    "references": [
+      {
+        "path": "./tsconfig.app.json"
+      }
+    ]
 }

--- a/packages/remix/src/generators/application/lib/update-unit-test-config.ts
+++ b/packages/remix/src/generators/application/lib/update-unit-test-config.ts
@@ -3,6 +3,7 @@ import {
   joinPathFragments,
   stripIndents,
   type Tree,
+  updateJson,
   workspaceRoot,
 } from '@nx/devkit';
 import {
@@ -55,6 +56,32 @@ export function updateUnitTestConfig(
         .replace('jest.preset.js', 'jest.preset.cjs')
     );
   }
+
+  const pathToTsConfigSpec = joinPathFragments(
+    pathToRoot,
+    `tsconfig.spec.json`
+  );
+
+  updateJson(tree, pathToTsConfigSpec, (json) => {
+    json.includes = [
+      'vite.config.ts',
+      'vitest.config.ts',
+      'app/**/*.ts',
+      'app/**/*.tsx',
+      'app/**/*.js',
+      'app/**/*.jsx',
+      'tests/**/*.spec.ts',
+      'tests/**/*.test.ts',
+      'tests/**/*.spec.tsx',
+      'tests/**/*.test.tsx',
+      'tests/**/*.spec.js',
+      'tests/**/*.test.js',
+      'tests/**/*.spec.jsx',
+      'tests/**/*.test.jsx',
+    ];
+
+    return json;
+  });
 
   return addDependenciesToPackageJson(
     tree,

